invenio_app_rdm/communities_ui/views/communities.py

```diff
@@ -89,3 +92,25 @@ def communities_home(pid_value, community, community_ui):
             permissions=permissions,
             records=records_ui,
         )
+
+
+@pass_community(serialize=True)
+def community_static_page(pid_value, community, community_ui):
+    """Community static page."""
+    permissions = community.has_permissions_to(
+        ["update", "read", "search_requests", "search_invites", "moderate"]
+    )
+
+    try:
+        page = current_pages_service.read_by_url(g.identity, request.path).to_dict()
+
+    except (PageNotFoundError, NoResultFound):
+        abort(404)
```

Comment:
major: we have error handlers defined in [records-resources](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/resources/errors.py), i.e. you can use a `NoResultFound` [handler](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/resources/errors.py#L152).